src/library.cpp

```diff
@@ -79,6 +86,9 @@ bool Library::addBook(const Book& book)
   updateBookDB(book);
   try {
     auto& oldbook = m_books.at(book.getId());
+    if ( ! booksReferToTheSameArchive(oldbook, book) ) {
+      dropReader(book.getId());
+    }
```

Comment:
Why do you think so? Only those books for which the path of the ZIM file has changed will be dropped from the cache (and I don't see why it would occur in practice at all).